### ISSUE TITLE
An error when I test Bytetrack

### ISSUE THREAD

**Issue by ZJZhao123** [open-mmlab/mmtracking#772]
The error is as follows.

load checkpoint from local path: checkpoint/bytetrack_yolox_x_crowdhuman_mot17-private-half_20211218_205500-1985c9f0.pth
The model and loaded state dict do not match exactly.

I downloaded the model and checkpoint from github and didn't modify anything.

**Comment by dyhBUPT** [comment]
Hi, maybe you should show us details, did you use the newest code? which branch did you use? what command did you use?

**Comment by ZJZhao123** [comment]
I used
_git clone https://github.com/open-mmlab/mmtracking.git
cd mmtracking_
to download code(default master),

and used
_python ./tools/test.py /home/zzj/mmtracking/configs/mot/bytetrack/bytetrack_yolox_x_crowdhuman_mot17-private-halfno.py  --checkpoint /home/zzj/mmtracking/checkpoint/bytetrack_yolox_x_crowdhuman_mot17-private-half_20211218_205500-1985c9f0.pth --eval track_
to test.

The config is as bellow
__base_ = [
    '../../_base_/models/yolox_x_8x8.py',
    '../../_base_/datasets/mot_challenge.py', '../../_base_/default_runtime.py'
]

img_scale = (800, 1440)
samples_per_gpu = 1

model = dict(
    type='ByteTrack',
    detector=dict(
        input_size=img_scale,
        random_size_range=(18, 32),
        bbox_head=dict(num_classes=1),
        test_cfg=dict(score_thr=0.01, nms=dict(type='nms', iou_threshold=0.7)),
        init_cfg=dict(
            type='Pretrained',
            checkpoint=  # noqa: E251
            'https://download.openmmlab.com/mmdetection/v2.0/yolox/yolox_x_8x8_300e_coco/yolox_x_8x8_300e_coco_20211126_140254-1ef88d67.pth'  # noqa: E501
        )),
    motion=dict(type='KalmanFilter'),
    tracker=dict(
        type='ByteTracker',
        obj_score_thrs=dict(high=0.6, low=0.1),
        init_track_thr=0.7,
        weight_iou_with_det_scores=True,
        match_iou_thrs=dict(high=0.1, low=0.5, tentative=0.3),
        num_frames_retain=30))

train_pipeline = [
    dict(
        type='Mosaic',
        img_scale=img_scale,
        pad_val=114.0,
        bbox_clip_border=False),
    dict(
        type='RandomAffine',
        scaling_ratio_range=(0.1, 2),
        border=(-img_scale[0] // 2, -img_scale[1] // 2),
        bbox_clip_border=False),
    dict(
        type='MixUp',
        img_scale=img_scale,
        ratio_range=(0.8, 1.6),
        pad_val=114.0,
        bbox_clip_border=False),
    dict(type='YOLOXHSVRandomAug'),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Resize',
        img_scale=img_scale,
        keep_ratio=True,
        bbox_clip_border=False),
    dict(type='Pad', size_divisor=32, pad_val=dict(img=(114.0, 114.0, 114.0))),
    dict(type='FilterAnnotations', min_gt_bbox_wh=(1, 1), keep_empty=False),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=img_scale,
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[0.0, 0.0, 0.0],
                std=[1.0, 1.0, 1.0],
                to_rgb=False),
            dict(
                type='Pad',
                size_divisor=32,
                pad_val=dict(img=(114.0, 114.0, 114.0))),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='VideoCollect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=samples_per_gpu,
    workers_per_gpu=4,
    persistent_workers=True,
    train=dict(
        _delete_=True,
        type='MultiImageMixDataset',
        dataset=dict(
            type='CocoDataset',
            ann_file=[
                'data/MOT17/annotations/half-train_cocoformat.json',
                'data/crowdhuman/annotations/crowdhuman_train.json',
                'data/crowdhuman/annotations/crowdhuman_val.json'
            ],
            img_prefix=[
                'data/MOT17/train'
                , 'data/crowdhuman/train',
                'data/crowdhuman/val'
            ],
            classes=('pedestrian', ),
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='LoadAnnotations', with_bbox=True)
            ],
            filter_empty_gt=False),
        pipeline=train_pipeline),
    val=dict(
        pipeline=test_pipeline,
        interpolate_tracks_cfg=dict(min_num_frames=5, max_num_frames=20)),
    test=dict(
        pipeline=test_pipeline,
        interpolate_tracks_cfg=dict(min_num_frames=5, max_num_frames=20)))

# optimizer
# default 8 gpu
optimizer = dict(
    type='SGD',
    lr=0.001 / 2 * samples_per_gpu,
    momentum=0.9,
    weight_decay=5e-4,
    nesterov=True,
    paramwise_cfg=dict(norm_decay_mult=0.0, bias_decay_mult=0.0))
optimizer_config = dict(grad_clip=None)

# some hyper parameters
total_epochs = 80
num_last_epochs = 10
resume_from = None
interval = 5

# learning policy
lr_config = dict(
    policy='YOLOX',
    warmup='exp',
    by_epoch=False,
    warmup_by_epoch=True,
    warmup_ratio=1,
    warmup_iters=1,
    num_last_epochs=num_last_epochs,
    min_lr_ratio=0.05)

custom_hooks = [
    dict(
        type='YOLOXModeSwitchHook',
        num_last_epochs=num_last_epochs,
        priority=48),
    dict(
        type='SyncNormHook',
        num_last_epochs=num_last_epochs,
        interval=interval,
        priority=48),
    dict(
        type='ExpMomentumEMAHook',
        resume_from=resume_from,
        momentum=0.0001,
        priority=49)
]

checkpoint_config = dict(interval=1)
evaluation = dict(metric=['bbox', 'track'], interval=1)
search_metrics = ['MOTA', 'IDF1', 'FN', 'FP', 'IDs', 'MT', 'ML']

# you need to set mode='dynamic' if you are using pytorch<=1.5.0
mode = 'dynamic'
fp16 = dict(loss_scale=dict(init_scale=512.))_

Then I recieved
![image](https://user-images.githubusercontent.com/97434605/206409151-e6ea3241-b6e4-43eb-9808-ef869c34d620.png)

**Comment by dyhBUPT** [comment]
The keys `ema_xxx` is only used in training, thus the reported `The model and loaded state dict do not match exactly.` is not an error, and you can ignore it.

**Comment by dyhBUPT** [comment]
For your slightly lower metrics, HOTA 67.1 vs 67.7, did you use the latest version of mmtracking and mmdetection?

And, the mmtracking 1.x is recommended: https://github.com/open-mmlab/mmtracking/tree/dev-1.x

**Comment by dyhBUPT** [comment]
Glad to hear that~
Can you share the reason?

**Comment by ZJZhao123** [comment]
I understand your first comment and try your second comment.

I created a new environment with python 3.9 and mmtracking-dev-1.x and test with the code as follows
` ./tools/test.py /home/zzj/mmtracking-dev-1.x/configs/mot/bytetrack/bytetrack_yolox_x_8xb4-80e_crowdhuman-mot17halftrain_test-mot17halfval.py  --checkpoint /home/zzj/mmtracking/checkpoint/bytetrack_yolox_x_crowdhuman_mot17-private-half_20211218_205500-1985c9f0.pth
`
And the result is even worse
![image](https://user-images.githubusercontent.com/97434605/206656448-a8d40154-e7fa-45f1-8d3f-319d278c818c.png)

I used to use an environment with python 3.7 and mmtracking default.

**Comment by ZJZhao123** [comment]
I also used /bytetrack_yolox_x_8xb4-amp-80e_crowdhuman-mot17halftrain_test-mot17halfval.py to test, the result is better but still slightly lower.
![image](https://user-images.githubusercontent.com/97434605/206663043-9d0754f8-83a1-499b-8239-e8765493001c.png)

**Comment by dyhBUPT** [comment]
I've just tested ByteTrack (dev-1.x) and got similar results. There may be some problem.
![image](https://user-images.githubusercontent.com/99722489/208346773-95529cfa-780f-45c0-aaf2-bead1459c9f6.png)
